File: lib_network_retrofit/src/main/java/com/network/retrofit/interceptor/LoggingInterceptor.java
package com.network.retrofit.interceptor;

import android.util.Log;

import com.common.log.KLog;

import java.io.IOException;
import java.util.Locale;

import okhttp3.Interceptor;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.Response;
import okhttp3.ResponseBody;

/**
 * 日志打印
 */
public class LoggingInterceptor implements Interceptor {
    public static final String TAG = LoggingInterceptor.class.getSimpleName();
    @Override
    public Response intercept(Chain chain) throws IOException {

        Request request = chain.request();
        Log.v(TAG, "request:" + request.toString());
        long t1 = System.nanoTime();
        Response response = chain.proceed(chain.request());
        long t2 = System.nanoTime();
        Log.v(TAG, String.format(Locale.getDefault(), "Received response %s for %s in %.1fms%n%s",
                response.code(),response.request().url(), (t2 - t1) / 1e6d, response.headers()));
        MediaType mediaType = response.body() != null ? response.body().contentType() : null;
        String content = response.body() != null ? response.body().string() : null;
//        Log.i(TAG, "response body:" + content);
//        Logger.json(content);
        KLog.json(content);
        return response.newBuilder().body(ResponseBody.create(mediaType, content != null ? content : "")).build();

    }
}

File: lib_network_retrofit_coroutine/src/main/java/com/network/retrofit_coroutine/LoggingInterceptor.kt
package com.network.retrofit_coroutine

import android.util.Log
import com.common.log.KLog
import okhttp3.Interceptor
import okhttp3.Response
import okhttp3.ResponseBody
import java.util.Locale

class LoggingInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        KLog.d("request:$request")
        val t1 = System.nanoTime()
        val response = chain.proceed(chain.request())
        val t2 = System.nanoTime()
        KLog.d(
            String.format(
                Locale.getDefault(), "Received response %s for %s in %.1fms%n%s",
                response.code, response.request.url, (t2 - t1) / 1e6, response.headers
            )
        )
        val mediaType = if (response.body != null) response.body!!.contentType() else null
        val content = if (response.body != null) response.body!!.string() else null
        KLog.json(content)
        return response.newBuilder().body(ResponseBody.create(mediaType, content ?: "{}")).build()

    }
}